lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -749,7 +749,7 @@ struct StoreAsyncOpConversion
         typeConverter->convertType(rewriter.getI8Type()), 3);
 
     auto threadId = getThreadId(rewriter, loc);
-    Value pred = icmp_eq(urem(threadId, i32_val(32)), i32_val(0));
```

Comment:
@goostavz is the change expected?

Reply:
In this case, all warps will do their own storing, so the predicate will always be true for all warps.